src/services/has-many-getter.js

```diff
@@ -1,134 +1,57 @@
-const _ = require('lodash');
-const P = require('bluebird');
-const Interface = require('forest-express');
-const orm = require('../utils/orm');
-const QueryBuilder = require('./query-builder');
-const SearchBuilder = require('./search-builder');
-const FiltersParser = require('./filters-parser');
-const CompositeKeysManager = require('./composite-keys-manager');
-const extractRequestedFields = require('./requested-fields-extractor');
-const Operators = require('../utils/operators');
+import CompositeKeysManager from './composite-keys-manager';
+import ResourcesGetter from './resources-getter';
 
-class HasManyGetter {
-  constructor(model, association, options, params) {
-    this.model = model;
-    this.association = association;
-    this.params = params;
-    this.queryBuilder = new QueryBuilder(model, options, params);
-    this.schema = Interface.Schemas.schemas[association.name];
-    [this.primaryKeyModel] = _.keys(model.primaryKeys);
-    this.operators = Operators.getInstance(options);
-    this.filtersParser = new FiltersParser(this.schema, params.timezone, options);
-    this.fieldNamesRequested = extractRequestedFields(
-      params.fields, association, Interface.Schemas.schemas,
-    );
-    this.searchBuilder = new SearchBuilder(
-      association,
-      options,
-      params,
-      this.fieldNamesRequested,
-    );
-  }
-
-  async buildWhereConditions({ associationName, search, filters }) {
-    const { AND } = this.operators;
-    const where = { [AND]: [] };
-
-    if (search) {
-      const searchCondition = this.searchBuilder.perform(associationName);
-      where[AND].push(searchCondition);
-    }
-
-
-    if (filters) {
-      const formattedFilters = await this.filtersParser.perform(filters);
-      where[AND].push(formattedFilters);
-    }
+class HasManyGetter extends ResourcesGetter {
+  constructor(model, association, lianaOptions, params) {
+    super(association, lianaOptions, params);
 
-    return where;
+    this._rootModel = model;
   }
 
-  async findQuery(queryOptions) {
-    if (!queryOptions) { queryOptions = {}; }
-    const { associationName, recordId } = this.params;
+  async _getRecords() {
+    const { associationName, recordId } = this._params;
+    const options = await this._buildQueryOptions();
 
-    const where = await this.buildWhereConditions(this.params);
-    const include = this.queryBuilder.getIncludes(this.association, this.fieldNamesRequested);
+    const record = await this._rootModel.findOne({
+      // Don't fetch parent attributes (perf)
+      attributes: [],
 
-    const record = await orm.findRecord(this.model, recordId, {
-      order: queryOptions.order,
+      // We are ordering with the relation
+      // https://github.com/sequelize/sequelize/issues/4553#issuecomment-213989980
+      order: options.order.map((o) => [this._model, ...o]),
```

Comment:
it worked before because the query-builder.order function was way more complicated than the query-options.order.
This just tells sequelize where the column is

I did this, because I don't want the QueryOption class to know what a HasMany association is (it's too complex already)